Add more components to Snaps Custom UI (#1835)
MetaMask/metamask-docs: snaps/features/custom-ui/index.md

Outputs a formatted text field for a blockchain address. 
The address is automatically displayed with a [Jazzicon](https://www.npmjs.com/package/@metamask/jazzicon)
and truncated value. 

> When the address is truncated, hovering over the address shows the full value in a tooltip.

Reviewer: ```suggestion
```
This isn't true

Author: Oops! So hovering always shows the full value?

Reviewer: We don't have the tooltip anymore at all 😅

We are considering other ways to provide the full address when truncated

#### Props 

- `address`: `string` - A valid Ethereum address, starting with `0x`, or a valid 
  [CAIP-10](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-10.md) address.

> - `truncate`: `boolean` - (Optional) whether to truncate the address. Defaults to `true`.
> - `displayName`: `boolean` - (Optional) whether to display the internally saved account label in place of the address.
>   Defaults to `true`.
> - `avatar`: `boolean` - (Optional) whether to show the address Jazzicon. Defaults to `true`.

Reviewer 2: ```suggestion
- `truncate`: `boolean` - (Optional) Whether to truncate the address. The default is `true`.
- `displayName`: `boolean` - (Optional) Whether to display the internally saved account label in place of the address.
  The default is `true`.
- `avatar`: `boolean` - (Optional) Whether to show the address Jazzicon. The default is `true`.
```

Outputs a [Jazzicon](https://www.npmjs.com/package/@metamask/jazzicon) for an address.

#### Props

- `address`: `string` - A valid [CAIP-10](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-10.md) address.

> - `size`: `string` - (Optional) The size of the avatar. Can be `sm`, `md`, or `lg`. Defaults to `md`.

Reviewer 2: ```suggestion
- `size`: `string` - (Optional) The size of the avatar. Possible values are `"sm"`, `"md"`, and `"lg"`. The default is `"md"`.
```

});
```

> ### `Banner`

Reviewer 2: Can you move this section up before the `Bold` component?

Author: Will do!

> Outputs a banner alert component that can be used to display important messages with different severity levels.

Reviewer 2: ```suggestion
Outputs a banner that can be used to display important messages with different severity levels.
```

#### Props

> - `title`: `string` - Title of the banner.
> - `severity`: `"danger" | "info" | "success" | "warning"` - Severity level of the banner.
> - `children`: The content to display in the banner. Can include:
>   - Text components
>   - Formatting components
>   - Links
>   - Icons
>   - Buttons
>   - Skeleton loaders

Reviewer 2: ```suggestion
- `title`: `string` - The title of the banner.
- `severity` - The severity level of the banner.
  Possible values are `"danger"`, `"info"`, `"success"`, and `"warning"`.
- `children` - The content to display in the banner.
  This can include text, formatting components, [`Link`](#link) components, [`Icon`](#icon) components, [`Button`](#button) components, and skeleton loaders.
```

Questions: What do you mean by "formatting components"? Is it bold and italic text? Also, are "skeleton loaders" added using the `Image` component?

Author: Yes formatting components is bold and italic. Skeleton loaders -- will get back to you about that.

Author: Fixed :)

- `alignment` - (Optional) The alignment of the elements inside the box.
  Possible values are `"start"`, `"center"`, `"end"`, `"space-between"`, and `"space-around"`.
  The default is `"start"`.

> - `center`: `boolean` - (Optional) whether to center the children within the box.
>   Defaults to `false`.

Reviewer 2: ```suggestion
- `center`: `boolean` - (Optional) Whether to center the children within the box.
  The default is `false`.
```

<img src={require("../../assets/custom-ui-checkbox.png").default} alt="Checkbox UI example" width="450px" style={{border: "1px solid #DCDCDC"}} />
</p>

### `Container`

Outputs a container component that can hold a box of content and an optional footer. This is useful for creating structured layouts with action buttons at the bottom in [custom dialogs](dialogs.md#display-a-custom-dialog).

#### Props

> - `backgroundColor`: `"default" | "alternative"` - (optional) The background color of the container. Defaults to `"default"`.
> - `children`: Can be either:
>   - A single Box element, or
>   - An array containing a Box element and a Footer element

Reviewer 2: ```suggestion
- `backgroundColor` - (Optional) The background color of the container.
  Possible values are `"default"` and `"alternative"`.
  The default is `"default"`.
- `children`: The contents of the container.
  This can be a single [`Box`](#box) component, or an array containing a [`Box`](#box) component and a [`Footer`](#footer) component.
```

<img src={require("../../assets/custom-ui-file-input.png").default} alt="File input UI example" width="450px" style={{border: "1px solid #DCDCDC"}} />
</p>

### `Footer`

> Outputs a footer component that can contain one or two buttons. This component is typically used within a [`Container`](#container) to create action buttons at the bottom of a dialog or panel.

Reviewer 2: ```suggestion
Outputs a footer that can contain one or two buttons. This component is typically used within a [`Container`](#container) to create action buttons at the bottom of a dialog or panel.
```

#### Props

> - `children`: Can be either one or two [`Button`](#button) components.

Reviewer 2: ```suggestion
- `children` - The contents of the footer.
  This can be one or two [`Button`](#button) components.
```

<img src={require("../../assets/custom-ui-selector.png").default} alt="Selector UI example" width="450px" />
</p>

### `Skeleton`

Outputs an animated loading container.

#### Props

> - `width`: `Array<number | string>` - (Optional) width of the skeleton.
> - `height`: `Array<number | string>` - (Optional) height of the skeleton.
> - `borderRadius`: `string` - (Optional) radius of the corners, can be `none`, `medium` or `full`.

Reviewer 2: ```suggestion
- `width`: `Array<number | string>` - (Optional) The width of the skeleton.
- `height`: `Array<number | string>` - (Optional) The height of the skeleton.
- `borderRadius`: `string` - (Optional) The radius of the corners. Possible values are `"none"`, `"medium"` or `"full"`.
```

> - `borderRadius`: `string` - (Optional) The radius of the corners. Possible values are `"none"`, `"medium"` or `"full"`.

Reviewer 2: Since these are all optional, do they have default values?

Author: Added!

- `alignment` - (Optional) The alignment of the text.
  Possible values are `"start"`, `"center"`, and `"end"`.
  The default is `"start"`.
- `children` - The content to display.

>   This can include strings, [`Bold`](#bold), [`Italic`](#italic), [`Icon`](#icon), [`Link`](#link), and [`Skeleton`](#skeleton) components.

Reviewer 2: ```suggestion
  This can include strings, and [`Bold`](#bold), [`Italic`](#italic), [`Icon`](#icon), [`Link`](#link), and [`Skeleton`](#skeleton) components.
```

<img src={require("../../assets/custom-ui-tooltip.png").default} alt="Tooltip UI example" width="450px" style={{border: "1px solid #DCDCDC"}} />
</p>

### `Value`

> Outputs a component that displays two text values side by side. This component can only be used as a child of the [`Row`](#row) component.

Reviewer 2: Can you update the `children` prop of the `Row` component to include `Value` as an option?

#### Props

> - `value`: `string | TextElement` - The value shown on the right side.
> - `extra`: `string | TextElement` - The extra text shown on the left side.

Reviewer 2: ```suggestion
- `value` - The value shown on the right side.
  This can be a string or a [`Text`](#text) component.
- `extra` - The extra text shown on the left side.
  This can be a string or a [`Text`](#text) component.
```
